plugin/cassandra/plugin.go

```diff
@@ -0,0 +1,666 @@
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_host",
+				Type:     "string",
+				Title:    "Cassandra Host",
+				Help:     "The hostname or IP address of your Cassandra server.",
+				Default:  "127.0.0.1",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_port",
+				Type:     "port",
+				Title:    "Cassandra Port",
+				Help:     "The 'native transport' TCP port that Cassandra server is bound to, listening for incoming connections.",
+				Default:  "9042",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_user",
+				Type:     "string",
+				Title:    "Cassandra Username",
+				Help:     "Username to authenticate to Cassandra as.",
+				Default:  "cassandra",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_password",
+				Type:     "password",
+				Title:    "Cassandra Password",
+				Help:     "Password to authenticate to Cassandra as.",
+				Default:  "cassandra",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_keyspace",
+				Type:     "string",
+				Title:    "Keyspace to Backup",
+				Help:     "The name of the keyspace to backup.",
+				Example:  "system",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_bindir",
+				Type:     "abspath",
+				Title:    "Path to Cassandra bin/ directory",
+				Help:     "The absolute path to the bin/ directory that contains the `nodetool` and `sstableloader` commands.",
+				Default:  "/var/vcap/packages/cassandra/bin",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_datadir",
+				Type:     "abspath",
+				Title:    "Path to Cassandra data/ directory",
+				Help:     "The absolute path to the data/ directory that contains the Cassandra database files.",
+				Default:  "/var/vcap/store/cassandra/data",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_tar",
+				Type:     "abspath",
+				Title:    "Path to `tar` utility",
+				Help:     "By default, the plugin will search the local `$PATH` to find the `tar` utility.",
+				Default:  DefaultTar,
+				Required: true,
+			},
+		},
+	}
+
+	plugin.Run(p)
+}
+
+type CassandraPlugin plugin.PluginInfo
+
+type CassandraInfo struct {
+	Host     string
+	Port     string
+	User     string
+	Password string
+	Keyspace string
+	BinDir   string
+	DataDir  string
+	Tar      string
+}
+
+// This function should be used to return the plugin's PluginInfo, however you decide to implement it
+func (p CassandraPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
+}
+
+// Called to validate endpoints from the command line
+func (p CassandraPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
+	var (
+		s    string
+		err  error
+		fail bool
+	)
+
+	s, err = endpoint.StringValueDefault("cassandra_host", "")
```

Comment:
As a matter of style, we are trying to move attribute names to be less namespaced.  For example, "host" is clearer than "cassandra_host" -- the original fear that people would just randomly change plugins without re-doing the config has proved to be unfounded.

Would you mind just dropping the `cassandra_` prefix on all configuration options?

Reply:
No prob., I'll do that.